server/service/osquery_test.go

```diff
@@ -572,6 +572,137 @@ func TestHostDetailQueries(t *testing.T) {
 	assert.Equal(t, "select foo", queries[hostAdditionalQueryPrefix+"foobar"])
 }
 
+func TestQueriesAndHostFeatures(t *testing.T) {
+	ds := new(mock.Store)
+	team1 := fleet.Team{
+		ID: 1,
+		Config: fleet.TeamConfig{
+			Features: fleet.Features{
+				EnableHostUsers:         true,
+				EnableSoftwareInventory: false,
+			},
+		},
+	}
+
+	team2 := fleet.Team{
+		ID: 2,
+		Config: fleet.TeamConfig{
+			Features: fleet.Features{
+				EnableHostUsers:         false,
+				EnableSoftwareInventory: true,
+			},
+		},
+	}
+
+	host := fleet.Host{
+		ID:       1,
+		Platform: "darwin",
+		NodeKey:  "test_key",
+		Hostname: "test_hostname",
+		UUID:     "test_uuid",
+		TeamID:   nil,
+	}
+
+	ds.AppConfigFunc = func(ctx context.Context) (*fleet.AppConfig, error) {
+		fmt.Println("ds.AppConfigFunc")
+		return &fleet.AppConfig{
+			Features: fleet.Features{
+				EnableHostUsers:         false,
+				EnableSoftwareInventory: false,
+			},
+		}, nil
+	}
+
+	ds.TeamFeaturesFunc = func(ctx context.Context, id uint) (*fleet.Features, error) {
+		fmt.Println("ds.TeamFeaturesFunc")
+		switch id {
+		case uint(1):
+			fmt.Println("1")
+			return &team1.Config.Features, nil
+		case uint(2):
+			fmt.Println("2")
+			return &team2.Config.Features, nil
+		default:
+			fmt.Println("default")
+			return nil, errors.New("team not found")
+		}
+	}
+
+	ds.LabelQueriesForHostFunc = func(ctx context.Context, host *fleet.Host) (map[string]string, error) {
+		return map[string]string{}, nil
+	}
+
+	ds.PolicyQueriesForHostFunc = func(ctx context.Context, host *fleet.Host) (map[string]string, error) {
+		return map[string]string{}, nil
+	}
+
+	lq := live_query_mock.New(t)
+	lq.On("QueriesForHost", uint(1)).Return(map[string]string{}, nil)
+	lq.On("QueriesForHost", uint(2)).Return(map[string]string{}, nil)
+	lq.On("QueriesForHost", nil).Return(map[string]string{}, nil)
+
+	t.Run("free license", func(t *testing.T) {
+		license := &fleet.LicenseInfo{Tier: fleet.TierFree}
+		svc := newTestService(t, ds, nil, lq, &TestServerOpts{License: license})
+
+		ctx := hostctx.NewContext(context.Background(), &host)
+		queries, _, _, err := svc.GetDistributedQueries(ctx)
+		require.NoError(t, err)
+		require.NotContains(t, queries, "fleet_detail_query_users")
+		require.NotContains(t, queries, "software_macos")
```

Comment:
Missing prefix in these, `fleet_detail_query_*`, right?

Reply:
you're absolutely right, thank you